server/coordinator.go

```diff
@@ -35,54 +35,60 @@ type coordinator struct {
 	ctx    context.Context
 	cancel context.CancelFunc
 
-	cluster *clusterInfo
-	opt     *scheduleOption
-	checker *replicaCheckController
-
-	schedulers map[string]*ScheduleController
-	operators  map[ResourceKind]map[uint64]Operator
+	cluster    *clusterInfo
+	opt        *scheduleOption
+	limiter    *scheduleLimiter
+	checker    *replicaChecker
+	operators  map[uint64]Operator
+	schedulers map[string]*scheduleController
 
 	histories *lruCache
 	events    *fifoCache
 }
 
 func newCoordinator(cluster *clusterInfo, opt *scheduleOption) *coordinator {
-	c := &coordinator{
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
the context is the root context for the coordinator itself, I think it is ok here.